app/assets/templates/forms/update.html

```diff
@@ -1,32 +1,34 @@
 <navbar></navbar>
-<div class="wrap-dashboard">
+<div class="wrap-dashboard" id="editForm">
   <div class="container-fluid wrap">
+    
+    <div class="container" id="formTitle">
+      <div class="row">
+        <h4>{{ name }}</h4>
+      </div>
+    </div>
+
     <div class="row">
       <div class="list-forms text-center">
-        <div class="col-xs-4">
-          <h4>Selections</h4>
-          <h6><a href="" ng-click="add('text')">Text</a></h6>
-          <h6><a href="" ng-click="add('textarea')">Large Text</a></h6>
-          <h6><a href="" ng-click="add('check')">Checkboxes</a></h6>
-          <h6><a href="" ng-click="add('radio')">Radioboxes</a></h6>
-          <h6><a href="" ng-click="add('dropdown')">Dropdown</a></h6>
-          <h6><a href="" ng-click="add('description')">Description</a></h6>
+        <div class="col-xs-2" id="sidebar">
+          <ul id="selections">
+            <li ng-repeat="category in categories" ng-click="add(category)" class="btn btn-embossed btn-primary">{{ category }}<i class="addContent fa fa-plus-square"></i></li>
+          </ul>
         </div>
-        <div class="col-xs-4">
-          <h4>{{ name }}</h4>
-          <ul>
-            <li ng-repeat="field in fields track by $index" ng-mouseover="showButtons = true" ng-mouseleave="showButtons = false">
-              <button ng-click="delete(field,$index)" ng-show="!editPanelVisible && showButtons">X</button>
+
+        <div class="col-xs-6" id="listOfContents">
+          <ul class="contentField">
+            <li ng-repeat="field in fields | orderFieldBy:'index'" ng-mouseover="showButtons = true" ng-mouseleave="showButtons = false" class="field btn btn-embossed btn-primary">
```

Comment:
orderBy only works with arrays,
I want to order by 'index' which is not the order of an array but a value that we allocate (see schema.rb) to keep track of the position. So I need to filter by the key index.